packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2126,10 +2128,9 @@ export class MergeTree {
                 if (matchedStart) {
                     if (!_node.isLeaf()) {
                         const childBlock = _node;
-                        go = this.nodeMap(childBlock, actions, 0, UniversalSequenceNumber, this.collabWindow.clientId,
-                            undefined);
+                        go = this.walkAllSegments(childBlock, leafAction);
```

Comment:
There's also some `leftExcursion` logic that is only used on `insertAtReferencePosition`, and explicitly attempts to exclude removed segments. I'm not sure if this should be adjusted as well.

Reply:
we probably will want it adjusted, as the behavior will change in my other pr, but i don't think we need to do that now